hw/ip_templates/racl_ctrl/racl_ctrl.core.tpl

```diff
@@ -4,6 +4,8 @@ CAPI=2:
 # SPDX-License-Identifier: Apache-2.0
 name: ${instance_vlnv(f"lowrisc:ip:{module_instance_name}:0.1")}
 description: "RACL Control permission IP"
+virtual:
```

Comment:
Why is racl_ctrl made a virtual core?

Reply:
The DV code gets copied via ipgen for explicitly for this racl_ctrl. So there is no need to a virtual core here.

Reply:
Sorry, I explained that rather badly. This line is saying that it *is* some version of the virtual core. Some non-templated testbench code that wants a `racl_ctrl` might depend on that without needing to know whether it's `darjeeling`, `earlgrey` or some other tea!

Reply:
I'm a bit confused here. Right now the testbench gets copied for a racl_ctrl in `darjeeling` or a different tea, right? So I am confused why it should get virtual. @a-will Can you also provide some fusesoc input here?

Reply:
We should avoid virtual VLNVs in templated cores. They make it impossible to have multiple ipgen parameterizations within a single design. (So far, I think we should limit virtual VLNV use to prims _only_.)

The testbench code can use the same `instance_vlnv()` and `module_instance_name` replacements as the RTL. This code is copied out, with replacements in templated DV code. In general, we do not want _some_ `racl_ctrl`. We want to test the specific generated form of the `racl_ctrl` template, which came from the same ipgen output.

Some related guidelines have more recently gone into ipgen docs, but maybe we need to be more explicit about DV code :smile: : https://github.com/lowRISC/opentitan/blob/0fcb0d14312fdc65f4851b7f964c7567cc96e703/util/ipgen/README.md?plain=1#L117-L121

Reply:
I agree! This makes perfect sense and I've just pushed a version which does things in a less "clever" (or indeed stupid) manner.